### ISSUE TITLE
Running into an invisible worm results in weirdly behaving half worms riding in the vehicle.

### ISSUE THREAD

**Issue by PatrikLundell** [CleverRaven/Cataclysm-DDA#57580]
### Describe the bug

I was driving along in my vehicle when there suddenly was a smash and a half worm appeared in the vehicle, attacking my character while riding along in the vehicle. I managed to stop the vehicle and attack back. At some point in this process a second half worm appeared, but I think it was a single worm initially.
After killing the first half worm I attacked the second one several times without hitting it. As it didn't seem to attack, I put on armor, took a few failing swings, and then walked out of the vehicle. I made a new attempt to attack it when it was on the door frame, and managed to kill it there.

### Steps to reproduce

1. Drive straight over a hidden, and thus invisible, worm (of some unknown kind), killing it in the resulting crash.
2. Find that the spawned half worm(s) are spawned inside the vehicle rather than on/in the ground.
3. Fight the half worms (ideally after stopping the vehicle).
4. Find that one of them doesn't seem to either attack or be eligible to be attacked.

### Expected behavior

A collision and half worms spawned on/in the ground.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19043.1645 (21H1)
- Game Version: 30c8dc3 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]
+ JSON addition of recipe workshop 2 blacksmith recipe for wind mill
+ JSON "mod" hack to disable portal storms (as the PR for the mod that did this was rejected).

### Additional context

Might have to drive in a cardinal direction (I was driving straight south) for the monsters to spawn inside the vehicle. I've had a previous case where I've killed two graboids and the resulting half worms with the same vehicle without them spawning inside the vehicle, and I think that driving was at an angle. It's also possible that it matters whether you hit the monster with vehicle tiles that have interior space behind them (rather than walls).

**Comment by NetSysFire** [comment]
I tried to find some worms to test this and ended up driving into a mass of slimes near a slime pit. Them splitting also made them appear in the car, which is probably the same thing.

**Comment by PatrikLundell** [comment]
Good thinking. Those buggers are easier to find and use as test subjects.

**Comment by NetSysFire** [comment]
>I wonder if this can lead to eggs hatching critters into vehicles straddling the eggs as they hatch as well?

Quite probably. I would not be surprised if it worked. I am reminded by the mechanics used in #57351, newly spawned critters probably board the vehicle automatically.

**Comment by PatrikLundell** [comment]
I don't care much about a hatching chicken, but there's a bunch of really nasty insectoid monster I definitely don't want spawned inside my vehicle.

**Comment by NetSysFire** [comment]
Do not underestimate a newly hatched goose attacking you while you drive. Some geese broke my arm ingame once.

On the other hand, wouldn't any egg be crushed by the vehicle if it drives over it? That might come before it would end up under the vehicle, but the egg being crushed is not a guaranteed thing I'd imagine. The chance is pretty low but should still be considered, especially since you can get some surprise wasp eggs hidden in shrubs or patches of underbrush, which would make it an extra nasty surprise.

**Comment by PatrikLundell** [comment]
Geese are fairly nasty, but some of the insectoids are on a different level of scary.

You'd get the straddling case when the egg is near the central line of the vehicle. The wheels are generally along the edges, with at least one wheel free line along the center. However, my vehicle has the wheels in the center two lines to avoid them getting damaged when the vehicle is attacked or runs into something. This also means the half worms teleported horizontally as well as upwards, since there are wheels under those tiles (the row behind the front row seats), with no wheels under the front row (rams).

Item crushing doesn't seem to happen if a vehicle is light enough (bicycle, although I might have had things crushed with those when both the basket and my character are fully loaded loaded up with loot). The MC crushes things, for instance.
It ought to be dependent on the pressure the weight per wheel contact area, but might have been simplified to just weight. The heavier the vehicle the more crushing it exerts, though (again, I haven't done any tests so I don't know if the number of wheels is a factor, but a fully loaded vehicle crushes more than an empty one).

**Comment by toweroglass** [comment]
Seems like it would be a 1 in a quintillion chance you would drive over a hatching egg seeing as the most common occurrence is they would hatch as you drive into the reality bubble.

**Comment by PatrikLundell** [comment]
I agree driving over is unlikely, but no less frustrating if it happens. I've set up an experiment where I've place a chicken egg under a stationary vehicle with closed doors, however.
